src/main/java/utilityclasses/DataValidator.java

```diff
@@ -0,0 +1,35 @@
+package utilityclasses;
+
+import lombok.NonNull;
+
+import java.time.LocalDate;
+
+public class DataValidator {
+
+    public static void validateString(String text) {
+        validateStringNotEmpty(text);
+    }
+
+    public static void validateDate(@NonNull LocalDate date) {
+        validateDateIsNotInFuture(date);
+    }
+
+    private static void validateStringNotEmpty(String text) {
+        if (text.isEmpty()) {
+            throw new IllegalArgumentException("The value can't be an empty string.");
+        }
+        validateStringNotBlank(text);
```

Comment:
Why do you check separately whether the string is empty and blank when isBlank() does both for you?